[feat] Add CoverageChecker to resizeIfNeeded

Zodbigt opened the pull request:
This PR adds coverage checking for the function 'resizeIfNeeded' in io/BufByteWriterImpl.scala. Resolves #56 

[Issue: #56]

Zodbigt committed:
[feat] Add CoverageChecker to resizeIfNeeded

Coverage checking has been added to the function 'resizeIfNeeded' in io/BufByteWriterImpl.scala

[Issue: #56]

stevensdavid commented on util-core/src/main/scala/com/twitter/io/BufByteWriterImpl.scala:
```diff
@@ -273,36 +274,47 @@ private final class DynamicBufByteWriter(arr: Array[Byte]) extends AbstractBufBy
+      if (size < 0 || requiredSize > MaxBufferSize) {
+        CoverageChecker.reached(funcName, 3)
         size = requiredSize
+      }
```
It might be good to check the else case here

Zodbigt committed:
[fix] Add one more code cov for resizeIfNeeded

This commit adds one more code cov for the resizeIfNeeded method to catch one more branch per feedback.

[issue: #56]
